src/generator/templates/api/lib/mbuild.mako

```diff
@@ -706,14 +706,14 @@ else {
 
         loop {
             % if default_scope:
-            let token = match ${auth_call}.token(&self.${api.properties.scopes}.keys().collect::<Vec<_>>()[..]).await {
-                Ok(token) => token.clone(),
-                Err(err) => {
-                    match  dlg.token(&err) {
+            let token = match ${auth_call}.get_token(&self.${api.properties.scopes}.keys().map(String::as_str).collect::<Vec<_>>()[..]).await {
+                Some(token) => token.clone(),
```

Comment:
As nice as it would be to adapt the return value to the new trait, I would feel much more comfortable to _not_ make this a breaking change.

This means `get_token()` has to return `Result<Option<String>, yup_oauth2::Error` which should then be handled as before. Those custom implementations simply don't use it, and those who have errors can find one to use in `yup_oauth2::Error`. This also means that `yup-oauth2` is still a required dependency.

It would be fair to mark this as 'TODO' since in the next major release it should be cleaned up to be like it is in this PR.